Sources/Basics/FileSystem/Path+Extensions.swift

```diff
@@ -41,7 +41,7 @@ extension AbsolutePath {
 
     /// Unlike ``AbsolutePath//extension``, this property returns all characters after the first `.` character in a
     /// `````. If no dot character is present in the filename or first dot is the last character, `nil` is returned.
-    var allExtensions: String? {
+    var allExtensions: [Substring]? {
```

Comment:
I think String is more appropriate here. we can cast.

Reply:
I thought that holding on to `Substring` is fine since the original string is present as a property of `AbsolutePath` anyway, but I don't mind changing to `String`, which was done in the latest commit